### PR TITLE
Upgrade Node to LTS v12.22.7

### PR DESCRIPTION
**Pull request by arelra**
## What does this change?

Upgrade Node from v10.22.0 to LTS v12.22.7

Refresh `yarn.lock` to get the latest dependency versions allowed by semver.

There's a lot of noise from Prettier so it's easier to look at the individual [commits](https://github.com/guardian/frontend/pull/24344/commits):

## Why?

**Node v10 is unsupported** and reached EOL 30 Apr 2021 : https://github.com/nodejs/Release
Node v12 is supported until it's EOL of 30 Apr 2022.
A step towards >= v14

## Errors

The following issues were encountered during the upgrade:
- `easing.spec` test failing due to increased precision of division
- `redplanet.spec` test failing due to incorrect use of promises (updated to use the same method as https://github.com/guardian/frontend/pull/24292 thank you @mxdvl )
- task runner failed silently when running webpack 😱 
  Required [listr](https://github.com/SamVerschueren/listr) upgrade (which brought in `rxjs6`)
- task runner failed when running tests.
  [`__tasks__/test/javascript/index.js`](https://github.com/guardian/frontend/blob/e8d8b42e5c64a21ab802612535c8f4c1c71f0caf/tools/__tasks__/test/javascript/index.js#L4) requires [any-observable's](https://github.com/sindresorhus/any-observable) `any-observable/register/rxjs-all` and hence expects `rxjs` to be installed.
  However this only happens as a side effect as it is not an explicit dependency.
  The version of `rxjs` following the `listr` upgrade is `rxjs6`.
  `rxjs6` required updating `__tasks__/test/javascript/index.js` to use the new [rxjs6 export locations](https://ncjamieson.com/avoiding-rxjs-compat/)
  `rxjs6` has been added as an explicit dependency in `package.json`
- task `compile` caused pages of deprecation warnings from:
    - deprecated svgo options
    ```
    "extendDefaultPlugins" utility is deprecated.
    Use "preset-default" plugin with overrides instead.
    For example:
    {
      name: 'preset-default',
      params: {
        overrides: {
          // customize plugin options
          convertShapeToPath: {
            convertArcs: true
          },
          // disable plugins
          convertPathData: false
        }
      }
    }
    ```
    - deprecated autoprefixer options
    ```   
    Replace Autoprefixer browsers option to Browserslist config.
    Use browserslist key in package.json or .browserslistrc file.
    
    Using browsers option can cause errors. Browserslist config 
    can be used for Babel, Autoprefixer, postcss-normalize and other tools.
    
    If you really need to use option, rename it to overrideBrowserslist.
    
    Learn more at:
    https://github.com/browserslist/browserslist#readme
    https://twitter.com/browserslist
    ```
- prettier errors

## Comparing Build Output

This section is just an fyi in case it is useful to future upgrades.

Because of the update to the configuration of [svgo](https://github.com/svg/svgo), `autoprefixer` and the task runner itself it was necessary to:

a) run tasks in verbose mode to capture any warning output
b) compare the generated output of tasks **against a known good Frontend build**, in particular:
  - SVG tasks
  - the task `compile` to make sure all assets types are generated

### Running tasks in verbose mode

To surface potential warning messages run tasks in verbose mode and capture stdout e.g.:

`./tools/task-runner/runner compile/inline-svgs --verbose --stdout`

### Comparing svgo output

Inspect any potential SVG regressions.

Using a diff tool to compare (with help from Dr.SVG @mxdvl ) the output of the below tasks:

| Task  | Input folder | Output folder |
| :---  | :---  | :---  |
| `compile/images`  | `/static/src/images/` | `/static/src/stylesheets/icons` |
| `compile/inline-svgs`  | `/static/src/inline-svgs`  | `/common/conf/assets/inline-svgs`  |

### Comparing Target output

After running the task `compile` check `/static/target` (ignoring the hash changes) to ensure all asset types are generated.
    
## Tested

- [x] Locally
- [x] On CODE (optional)

### CODE REVIEW

**Review by mxdvl — COMMENTED** [review]
Isn't `toBeCloseTo` in Jest better than a custom wrapper? 

https://jestjs.io/docs/using-matchers#numbers

**Comment by arelra** [comment]
> Isn't `toBeCloseTo` in Jest better than a custom wrapper?
> 
> https://jestjs.io/docs/using-matchers#numbers

TIL thanks!

File: static/src/javascripts/projects/commercial/modules/comment-adverts.spec.ts [hunk 1]
```diff
@@ -62,7 +62,7 @@ const refreshAdvert = refreshAdvert_;
 const mockHeight = (height: number) => {
 	// this is an issue with fastdom's typing of measure: () => Promise<void>
 	jest.spyOn(fastdom, 'measure').mockReturnValue(
-		(Promise.resolve(height) as unknown) as Promise<void>,
+		Promise.resolve(height) as unknown as Promise<void>,
```
Comment (mxdvl) [inline static/src/javascripts/projects/commercial/modules/comment-adverts.spec.ts:65]: This is obviously wrong, isn’t it?

`Promise<number>` can’t also be `Promise<void>`.

File: static/src/javascripts/projects/commercial/modules/dfp/redplanet.js [hunk 2]
```diff
@@ -28,29 +28,37 @@ const initialise = () => {
...
-			return import(
-				/* webpackChunkName: "redplanet" */ '../../../../lib/launchpad.js'
-			).then(() => {
-				initialise();
-				return Promise.resolve();
-			});
+        // CCPA only runs in the US and tcfv2 outside Aus
+		// Redplanet only runs in Australia so this should never happen
+		if (!state.aus) {
+			rejectRedPlanetLoaded('unknown framework');
```
Comment (mxdvl) [inline static/src/javascripts/projects/commercial/modules/dfp/redplanet.js:48]:
```suggestion
			rejectRedPlanetLoaded('Redplanet should only run in Australia on AUS mode');
```

File: static/src/javascripts/projects/commercial/modules/dfp/redplanet.spec.js [hunk 3]
```diff
@@ -139,9 +139,7 @@ describe('init', () => {
 		isInAuOrNz.mockReturnValue(true);
 		onConsentChange.mockImplementation(CcpaWithConsentMock);
 		getConsentFor.mockReturnValue(true);
-		expect(await init).toThrow(
-			`Error running Redplanet without AUS consent. It should only run in Australia on AUS mode`,
-		);
+		await expect(init()).rejects.toEqual('unknown framework');
```
Comment (mxdvl) [inline static/src/javascripts/projects/commercial/modules/dfp/redplanet.spec.js:142]:
```suggestion
		await expect(init()).rejects.toMatch(/Australia in AUS/);
```

**Review by mxdvl — APPROVED** [review]
Excellent work! 🙌

**Comment by mxdvl** [comment]
Sorry, #24349 introduced some changes in files touched here. Hopefully it’s easy to resolve the conflicts.

**Pull request merged**